hal/src/rtl872x/usart_hal.cpp

```diff
@@ -461,7 +461,10 @@ class Usart {
                     uart->receiving_ = false;
                     uint8_t temp[MAX_UART_FIFO_SIZE];
                     uint32_t inFifo = UART_ReceiveDataTO(uartInstance, temp, MAX_UART_FIFO_SIZE, 1);
-                    uart->rxBuffer_.put(temp, inFifo);
+                    const ssize_t canWrite = uart->rxBuffer_.space();
```

Comment:
This doesn't look right, we should problably:
```suggestion
                    const ssize_t canWrite = uart->rxBuffer_.space();
                    if (canWrite > 0) {
                        uint8_t temp[MAX_UART_FIFO_SIZE];
                        uint32_t inFifo = UART_ReceiveDataTO(uartInstance, temp, std::min<size_t>(canWrite, MAX_UART_FIFO_SIZE), 1);
                        uart->rxBuffer_.put(temp, inFifo);
                    }
```